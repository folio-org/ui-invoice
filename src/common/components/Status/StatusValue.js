import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { KeyValue } from '@folio/stripes/components';
import { VersionKeyValue } from '@folio/stripes-acq-components';

import { getInvoiceStatusLabel } from '../../utils';

const StatusValue = ({ isVersionView, name, value }) => {
  const KeyValueComponent = isVersionView ? VersionKeyValue : KeyValue;

  return (
    <KeyValueComponent
      label={<FormattedMessage id="ui-invoice.invoice.details.information.status" />}
      name={name}
    >
      <FormattedMessage id={getInvoiceStatusLabel({ status: value })} />
    </KeyValueComponent>
  );
};

StatusValue.propTypes = {
  isVersionView: PropTypes.bool,
  name: PropTypes.string,
  value: PropTypes.string.isRequired,
};

export default StatusValue;
