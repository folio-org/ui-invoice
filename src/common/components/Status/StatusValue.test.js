import { render } from '@folio/jest-config-stripes/testing-library/react';

import StatusValue from './StatusValue';

jest.mock('../../utils', () => ({
  getInvoiceStatusLabel: jest.fn().mockReturnValue('getInvoiceStatusLabel util'),
}));

const renderStatusValue = (props) => (render(
  <StatusValue {...props} />,
));

describe('StatusValue', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should display status value', () => {
    const { asFragment } = renderStatusValue({ value: 'Approved' });

    expect(asFragment()).toMatchSnapshot();
  });
});
