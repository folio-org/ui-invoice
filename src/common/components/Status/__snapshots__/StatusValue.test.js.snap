// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusValue should display status value 1`] = `
<DocumentFragment>
  <div
    class="kvRoot"
  >
    <div
      class="kvLabel"
    >
      ui-invoice.invoice.details.information.status
    </div>
    <div
      class="kvValue"
      data-test-kv-value="true"
    >
      getInvoiceStatusLabel util
    </div>
  </div>
</DocumentFragment>
`;
